process/asyncExecution/executionTrack/executionResultsTracker.go

```diff
@@ -246,6 +260,35 @@ func (est *executionResultsTracker) SetLastNotarizedResult(executionResult data.
 	return nil
 }
 
+// RemoveByHash will remove the execution result by header hash
+func (est *executionResultsTracker) RemoveByHash(hash []byte) error {
+	est.mutex.Lock()
+	defer est.mutex.Unlock()
+
+	executionResult, found := est.executionResultsByHash[string(hash)]
+	if !found {
+		est.hashToRemoveOnAdd = hash
+		return nil
+	}
+
+	delete(est.executionResultsByHash, string(hash))
+	est.nonceHash.removeByNonce(executionResult.GetHeaderNonce())
```

Comment:
can you do a check that the removed nonce/hash is the last available or pending result and log an error otherwise?

we should not be able to remove intermediate blocks